edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/AuthChainHelpers.cs

```diff
@@ -27,6 +27,18 @@ public static Option<string> GetAuthTarget(Option<string> authChain)
             return authChainIds.FirstOption(id => true);
         }
 
+        public static Option<string> GetAuthParent(string targetId, Option<string> authChain)
+        {
+            return authChain.Match(
+                chain =>
+                {
+                    string[] authChainIds = GetAuthChainIds(chain);
+                    // The auth target is second element after the target
+                    return authChainIds.FirstOption(id => id != targetId);
```

Comment:
This seems a bit loose, if you are making assumptions about how the chains should be, then it will be better to validate that. Something like 
if(authChainIds[0] == targetId) return authChainIds[1];

Reply:
I changed to return second if available, without passing the targetId. I think it should not validate here if target is first, just return the value if it's there.